### PR TITLE
PIMOB-926: Billing form component: country

### PR DESCRIPTION
**Pull request by maxim-nosov-cko**
## Issue

[PIMOB-926: Billing form component: country](https://checkout.atlassian.net/browse/PIMOB-926)

## Proposed changes

Billing form country picker component and screen implemented.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [x] Reviewers assigned
* [x] I have performed a self-review of my code and manual testing
* [x] Lint and unit tests pass locally with my changes
* [x] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if applicable)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you choose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: frames/src/main/java/com/checkout/frames/component/base/InputComponent.kt [hunk 1]
```diff
@@ -6,26 +6,6 @@ import androidx.compose.foundation.layout.wrapContentHeight
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/component/base/InputComponent.kt:9]: Are you also deleting old code as we go ?

Comment (maxim-nosov-cko) [reply frames/src/main/java/com/checkout/frames/component/base/InputComponent.kt:9]: It was a preview for InputComponent, for debug/testing.

File: frames/src/main/java/com/checkout/frames/mapper/ImageStyleToDynamicComposableImageMapper.kt [hunk 2]
```diff
@@ -33,12 +35,18 @@ internal class ImageStyleToDynamicComposableImageMapper :
...
-                modifier = modifier,
-                painter = image,
-                contentDescription = "Dynamic Image",
-                colorFilter = style?.tinColor?.let { ColorFilter.tint(Color(it)) }
-            )
+            composableImage = @Composable {
+                Image(
+                    modifier = modifier,
+                    painter = image,
+                    contentDescription = "Dynamic Image",
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/mapper/ImageStyleToDynamicComposableImageMapper.kt:42]: Do we not put all the strings into one xml strings file ?

File: frames/src/main/java/com/checkout/frames/screen/countrypicker/CountryPickerViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    val searchCountries = mutableStateOf(allCountries)
+
+    val isSearchActive = mutableStateOf(false)
+    val goBack = mutableStateOf(false)
+
+    private var isInputFocused = false
+
+    fun onFocusChanged(isFocused: Boolean) {
+        isInputFocused = isFocused
+        if (isInputFocused) isSearchActive.value = true
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/screen/countrypicker/CountryPickerViewModel.kt:61]: Can this be just `isSearchActive.value = isInputFocused` ?

File: frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * 0x1F1E6 - Regional Indicator Symbol Letter A
+ */
+@SuppressWarnings("MagicNumber")
+internal fun Country.emojiFlag(): String {
+    if (this.iso3166Alpha2.length != 2) return ""
+
+    val firstCapitalizedChar = 'A'
+    val firstRegionalSymbol = 0x1F1E6
+    val iso2 = this.iso3166Alpha2.uppercase()
+    val firstLetter = iso2[0] - firstCapitalizedChar + firstRegionalSymbol
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt:22]: Does Kotlin array safe when using indices which are not valid ? iso2[0] and iso2[1] Do we need to check that iso2 always has something in it first.

File: frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * intended to be used to encode ISO 3166-1 alpha-2 two-letter country codes in a way
+ * that allows optional special treatment.
+ *
+ * A pair of regional indicator symbols is referred to as an emoji flag sequence
+ * (although it represents a specific region, not a specific flag for that region).
+ *
+ * 0x1F1E6 - Regional Indicator Symbol Letter A
+ */
+@SuppressWarnings("MagicNumber")
+internal fun Country.emojiFlag(): String {
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt:16]: Does this function directly have any unit tests ?

File: frames/src/main/java/com/checkout/frames/utils/extensions/ViewExtensions.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.checkout.frames.utils.extensions
+
+import android.graphics.Rect
+import android.view.View
+
+@SuppressWarnings("MagicNumber")
+internal fun View.isKeyboardOpen(): Boolean {
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/utils/extensions/ViewExtensions.kt:7]: Does this function directly have any unit tests ?

File: frames/src/main/res/drawable/cko_ic_cross_close.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="14dp"
+    android:height="14dp"
+    android:viewportWidth="14"
+    android:viewportHeight="14">
+  <path
+      android:pathData="M14,1.41L12.59,0L7,5.59L1.41,0L0,1.41L5.59,7L0,12.59L1.41,14L7,8.41L12.59,14L14,12.59L8.41,7L14,1.41Z"
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/res/drawable/cko_ic_cross_close.xml:7]: Question : What is this exactly ? what format is pathData in. Just curious to know.

**Review by deepesh-vasthimal-cko — APPROVED** [review]
Few minor comments on unit tests

File: frames/src/main/java/com/checkout/frames/utils/extensions/ViewExtensions.kt (hunk 6 above)
Comment (maxim-nosov-cko) [reply frames/src/main/java/com/checkout/frames/utils/extensions/ViewExtensions.kt:7]: This part is UI related, can't be tested with JUnit5 tests.

File: frames/src/main/res/drawable/cko_ic_cross_close.xml (hunk 7 above)
Comment (maxim-nosov-cko) [reply frames/src/main/res/drawable/cko_ic_cross_close.xml:7]: It's a common vector format.
Android Studio provides a possibility to transform svg vector image to xml vector image.

File: frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt (hunk 5 above)
Comment (maxim-nosov-cko) [reply frames/src/main/java/com/checkout/frames/utils/extensions/CountryExtensions.kt:16]: Discussed on stand-up.

**Pull request merged**